src/python/pants/backend/docker/parser.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import re
+from dataclasses import InitVar, dataclass, field
+from typing import Generator, Pattern
+
+from dockerfile import Command, parse_string
+
+from pants.backend.docker.target_types import DockerImageSources
+from pants.engine.fs import DigestContents, PathGlobs
+from pants.engine.rules import Get, collect_rules, rule
+from pants.option.global_options import GlobalOptions
+
+
+@dataclass(frozen=True)
+class DockerfileParseRequest:
+    sources: DockerImageSources
+
+
+@dataclass(frozen=True)
+class DockerfileInfo:
+    putative_target_addresses: tuple[str, ...] = ()
+
+
+@dataclass
+class DockerfileParser:
+    dockerfile: InitVar[str]
+    commands: tuple[Command, ...] = field(init=False)
+
+    pex_target_regexp: str = r"""
```

Comment:
It was being moved around and changed quite a bit, so haven't actually spent that much time reflecting on its final resting place. As it came from a global compiled regexp, and moving it from there, I moved the entire thing, instead of turning it into a string that was later compiled.
This way it keeps the file cleaner, but true, it doesn't need to be a field, perhaps that is weird and gives the wrong impression of being there for purposes of customization or some such.. 

Can move it out to global scope again.

Reply:
Yeah, right now there would be a separate member for each instance, which doesn't really make sense.